lib/middleware/proxy.js

```diff
@@ -34,7 +34,7 @@ function parseProxyConfig (proxies, config) {
 
     const hostname = proxyDetails.hostname || config.hostname
     const protocol = proxyDetails.protocol || config.protocol
-    const https = proxyDetails.protocol === 'https:'
+    const https = protocol === 'https:'
```

Comment:
The `proxyDetails` result from parsing the `proxyUrl`.  How then can `proxyDetails.protocol` be nullish? Seems to me that a better fix is to delete `|| config.protocol` and the variable `protocol` because it provides no value. Unless it does: what is the use case?

Reply:
Well. I can assure you that it can be nullish/falsy, because some tests that I have, were solved by this change.